File: calculadora.swift
import Foundation

func realizarOperacao(_ operacao: String, entre num1: Double, e num2: Double) -> Double? {
    switch operacao {
    case "+":
        return num1 + num2
    case "-":
        return num1 - num2
    case "*":
        return num1 * num2
    case "/":
        if num2 != 0 {
            return num1 / num2
        } else {
            print("Erro: Divisão por zero!")
            return nil
        }
    default:
        print("Operação inválida!")
        return nil
    }
}

// Solicitação de entrada do usuário
print("Digite o primeiro número:")
if let num1 = Double(readLine() ?? "") {
    print("Digite a operação (+, -, *, /):")
    if let operacao = readLine(), "+-*/".contains(operacao), operacao.count == 1 {
        print("Digite o segundo número:")
        if let num2 = Double(readLine() ?? "") {
            if let resultado = realizarOperacao(operacao, entre: num1, e: num2) {
                print("Resultado: \(resultado)")
            }
        } else {
            print("Número inválido!")
        }
    } else {
        print("Operação inválida!")
    }
} else {
    print("Número inválido!")
}

File: calculadora.c
#include <stdio.h>

int main() {
    float num1, num2, resultado;
    char operacao;

    // Solicitação de entrada do usuário
    printf("Digite o primeiro número: ");
    scanf("%f", &num1);

    printf("Digite a operação (+, -, *, /): ");
    scanf(" %c", &operacao);  // O espaço antes de %c ignora os espaços em branco

    printf("Digite o segundo número: ");
    scanf("%f", &num2);

    // Realiza a operação escolhida
    switch (operacao) {
        case '+':
            resultado = num1 + num2;
            break;
        case '-':
            resultado = num1 - num2;
            break;
        case '*':
            resultado = num1 * num2;
            break;
        case '/':
            if (num2 != 0) {
                resultado = num1 / num2;
            } else {
                printf("Erro: Divisão por zero!\n");
                return 1;  // Termina o programa com código de erro
            }
            break;
        default:
            printf("Operação inválida!\n");
            return 1;  // Termina o programa com código de erro
    }

    // Exibe o resultado
    printf("Resultado: %.2f\n", resultado);

    return 0;  // Termina o programa com sucesso
}

